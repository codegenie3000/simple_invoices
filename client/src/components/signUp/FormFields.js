import React from 'react';
import { Field } from 'redux-form';

import SignUpField from './SignUpField';

// Validations
const requiredValue = value => value ? undefined : '*Required';
const email = value => value && !/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(value) ?
    'Invalid email address' : undefined;
const minLength = value => value && value.length > 5 ? undefined : 'Must be 5 characters or more';

const formFields = [
    {
        name: 'displayName',
        label: 'Name',
        type: 'text',
        required: true,
        validate: requiredValue
    },
    {
        name: 'email',
        label: 'Email',
        type: 'email',
        required: true,
        validate: [ requiredValue, email ]
    },
    {
        name: 'password',
        label: 'Password',
        type: 'password',
        required: true,
        validate: [ requiredValue, minLength ]
    }
];

const Fields = props => {
    return formFields.map(function ({ name, label, type, required, validate }) {
        return (
            <Field
                key={ name }
                component={ SignUpField }
                name={ name }
                props={
                    {
                        label: label,
                        type: type,
                        required: required
                    }
                }
                validate={ validate }
            />
        );
    });
};
export default Fields;