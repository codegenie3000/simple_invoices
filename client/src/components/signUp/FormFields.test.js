import React from 'react';
import {shallow} from 'enzyme';

import FormFields from './FormFields';

describe('<FormFields />', ()=> {
    let component;

    beforeEach(()=> {
        component = shallow(<FormFields/>);
    });
    it('matches the snapshot', ()=> {
        expect(component).toMatchSnapshot();
    });
});