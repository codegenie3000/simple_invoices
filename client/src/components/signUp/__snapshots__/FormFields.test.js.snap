// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormFields /> matches the snapshot 1`] = `
Array [
  <Field
    component={[Function]}
    key="displayName"
    name="displayName"
    props={
      Object {
        "label": "Name",
        "required": true,
        "type": "text",
      }
    }
    validate={[Function]}
  />,
  <Field
    component={[Function]}
    key="email"
    name="email"
    props={
      Object {
        "label": "Email",
        "required": true,
        "type": "email",
      }
    }
    validate={
      Array [
        [Function],
        [Function],
      ]
    }
  />,
  <Field
    component={[Function]}
    key="password"
    name="password"
    props={
      Object {
        "label": "Password",
        "required": true,
        "type": "password",
      }
    }
    validate={
      Array [
        [Function],
        [Function],
      ]
    }
  />,
]
`;
